src/arbitrum/compounders/gmx/GlpCompounder.sol

```diff
@@ -75,15 +78,22 @@ contract GlpCompounder is TokenCompounderBase {
     /********************************** Mutated Functions **********************************/
 
     /// @notice Extending the base function to enable deposit of any one of GLP's underlying assets.
-    function depositUnderlying(address _underlyingAsset, uint256 _underlyingAssets, address _receiver, uint256 _minAmount) public nonReentrant returns (uint256 _shares) {
-        if (!(_underlyingAssets > 0)) revert ZeroAmount();
+    function depositUnderlying(address _underlyingAsset, uint256 _underlyingAmount, address _receiver, uint256 _minAmount) public payable nonReentrant returns (uint256 _shares) {
```

Comment:
What is the rational behind making the existing token based functions payable now, as opposed to having dedicated "ETH only" payable functions? Drawback I see here is that we mush everything together and make the dedicate functionalities more complex.

Reply:
We keep a simpler interface, easier to build on top of.